packages/video_player/video_player/android/src/main/java/io/flutter/plugins/videoplayer/Messages.java

```diff
@@ -0,0 +1,338 @@
+      return fromMapResult;
+    }
+  }
+
+  /** Generated class from Pigeon that represents data sent in messages. */
+  public static class LoopingMessage {
+    private Long textureId;
+    public Long getTextureId() { return textureId; }
+    public void setTextureId(Long setterArg) { this.textureId = setterArg; }
+
+    private Boolean isLooping;
+    public Boolean getIsLooping() { return isLooping; }
+    public void setIsLooping(Boolean setterArg) { this.isLooping = setterArg; }
+
+    HashMap toMap() {
+      HashMap<String, Object> toMapResult = new HashMap<String, Object>();
+      toMapResult.put("textureId", textureId);
+      toMapResult.put("isLooping", isLooping);
+      return toMapResult;
+    }
+    static LoopingMessage fromMap(HashMap map) {
+      LoopingMessage fromMapResult = new LoopingMessage();
+      fromMapResult.textureId = (int)map.get("textureId");
+      fromMapResult.isLooping = (bool)map.get("isLooping");
+      return fromMapResult;
+    }
+  }
+
+  /** Generated class from Pigeon that represents data sent in messages. */
+  public static class VolumeMessage {
+    private Long textureId;
+    public Long getTextureId() { return textureId; }
+    public void setTextureId(Long setterArg) { this.textureId = setterArg; }
+
+    private Double volume;
+    public Double getVolume() { return volume; }
+    public void setVolume(Double setterArg) { this.volume = setterArg; }
+
+    HashMap toMap() {
+      HashMap<String, Object> toMapResult = new HashMap<String, Object>();
+      toMapResult.put("textureId", textureId);
+      toMapResult.put("volume", volume);
+      return toMapResult;
+    }
+    static VolumeMessage fromMap(HashMap map) {
+      VolumeMessage fromMapResult = new VolumeMessage();
+      fromMapResult.textureId = (int)map.get("textureId");
+      fromMapResult.volume = (double)map.get("volume");
+      return fromMapResult;
+    }
+  }
+
+  /** Generated class from Pigeon that represents data sent in messages. */
+  public static class PositionMessage {
+    private Long textureId;
+    public Long getTextureId() { return textureId; }
+    public void setTextureId(Long setterArg) { this.textureId = setterArg; }
+
+    private Long position;
+    public Long getPosition() { return position; }
+    public void setPosition(Long setterArg) { this.position = setterArg; }
+
+    HashMap toMap() {
+      HashMap<String, Object> toMapResult = new HashMap<String, Object>();
+      toMapResult.put("textureId", textureId);
+      toMapResult.put("position", position);
+      return toMapResult;
+    }
+    static PositionMessage fromMap(HashMap map) {
+      PositionMessage fromMapResult = new PositionMessage();
+      fromMapResult.textureId = (int)map.get("textureId");
+      fromMapResult.position = (int)map.get("position");
+      return fromMapResult;
+    }
+  }
+
+  /** Generated interface from Pigeon that represents a handler of messages from Flutter.*/
+  public interface VideoPlayerApi {
```

Comment:
I have a feeling we don't have to do any proguarding of any sort currently since the user has to implement this interface and actually use all the generated classes at some point. Which is super awesome. Do you think that's right?

Reply:
Yep, should be the case.  There are instances where "toMap" or "fromMap" might not be used for a class if the communication is one directional.